packages/embarkjs/src/blockchain.js

```diff
@@ -367,6 +375,21 @@ Contract.prototype.send = function(value, unit, _options) {
   return this.blockchainConnector.send(options);
 };
 
+Contract.prototype.deployed = function() {
+  if (this.address === undefined && this._deployPromise === undefined) {
+    throw new Error("contract deployment hasn't happened yet");
+  }
+
+  if (this._deployPromise) {
+    return this._deployPromise;
+  }
+
+  const self = this;
+  return new Promise(function(resolve, _reject) {
+    resolve(self);
+  });
```

Comment:
The const self = this voodoo here can be avoided by using arrow functions (in fact, this can become a one liner):

```
return new Promise((resolve) => resolve(this));
```

And here the same expression with less code:

```
return Promise.resolve(this)
```

Reply:
I was Today Years Old when I learned this trick...